frontends/PyCDE/src/pycde/constructs.py

```diff
@@ -4,15 +4,58 @@
 
 from __future__ import annotations
 
-from .pycde_types import dim
+from .pycde_types import PyCDEType, dim
 from .value import BitVectorValue, ListValue, Value
 from circt.support import get_value
 from circt.dialects import msft, hw
 import mlir.ir as ir
 
+import types
 import typing
 
 
+def Wire(type: PyCDEType):
+  """Declare a wire. Used to create backedges. Must assign exactly once. Assign
+  method disappears after being called."""
+
+  def assign(self, new_value: Value):
+    if new_value.type != self.type:
+      raise TypeError(
+          f"Cannot assign {new_value.value.type} to {self.value.type}")
+
+    msft.replaceAllUsesWith(self.value, new_value.value)
+    self.value.owner.erase()
+    self.value = new_value.value
+    # Remove this method so it can't be called again.
+    del self.assign
```

Comment:
Wouldn't it be better to just throw a meaningful error instead of this?

Reply:
yeah, I thought I'd be fancy. I also think I should move it to `Value`, so that it shows up in code completions.